debug/trace.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package debug
+
+import (
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/topdown"
+)
+
+type stack interface {
```

Comment:
Should this be called `trace`? Also we don't need to make this public?

Reply:
There are two sides to this coin: from the perspective of topdown, this is a tracer; but from the view of the debugger, it behaves as a stack, so we're letting the naming reflect that perspective.
Currently, I don't think this needs to be public, but that could change in the future.